File: 0733-flood-fill/0733-flood-fill.js
/**
 * @param {number[][]} image
 * @param {number} sr
 * @param {number} sc
 * @param {number} color
 * @return {number[][]}
 */
var floodFill = function(image, sr, sc, color) {
    const [row,col] = [image.length, image[0].length];
    const color_same = image[sr][sc];
    function dfs (r,c) {
        if(r<0 || c<0 || r>=row || c>=col || image[r][c] !== color_same || image[r][c] === color) return;
        image[r][c] = color;
        dfs(r,c+1);
        dfs(r,c-1);
        dfs(r-1,c);
        dfs(r+1,c);
    }
    dfs(sr,sc);
    return image;
};

File: 0733-flood-fill/0733-flood-fill.cpp
class Solution {
public:
    void dfs(vector<vector<int>>& image, int r, int c, int color, int color_same){
        int rol = image.size(), col = image[0].size();
        if(r<0 || c<0 || r >= rol || c >= col || image[r][c] != color_same || image[r][c] == color) return;
        image[r][c] = color;
        dfs(image,r-1,c,color,color_same);
        dfs(image,r+1,c,color,color_same);
        dfs(image,r,c-1,color,color_same);
        dfs(image,r,c+1,color,color_same);
    }
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        dfs(image,sr,sc,color,image[sr][sc]);
        return image;
    }
};